subprojects/android-test/test-screenshot/src/main/java/com/avito/android/test/screenshot_test/internal/PermissionChecker.kt

```diff
@@ -0,0 +1,84 @@
+package com.avito.android.test.screenshot_test.internal
+
+import android.app.UiAutomation
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.ParcelFileDescriptor
+import androidx.core.content.ContextCompat
+import java.io.FileInputStream
+import java.io.IOException
+import java.io.InputStream
+import java.util.*
+
+
+class PermissionChecker(val context: Context) {
+
+    fun grantPermissions() {
+        val writePermission = ContextCompat.checkSelfPermission(context, WRITE_PERMISSION)
+        val readPermission = ContextCompat.checkSelfPermission(context, READ_PERMISSION)
+        println("request permissions")
+        for (permission in REQUIRED_PERMISSIONS) {
+            if (Build.VERSION.SDK_INT < 23) {
+                throw RuntimeException("We need $permission permission for screenshot tests")
+            }
+            val targetContext: Context = Registry.registry.instrumentation!!.targetContext
+            if (ContextCompat.checkSelfPermission(targetContext, permission) != PackageManager.PERMISSION_GRANTED) {
+                println("no $permission for target context")
+                grantPermission(targetContext, permission)
+            } else {
+                println("have $permission for target context")
+            }
+            if (ContextCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
+                println("no $permission for context")
+                grantPermission(context, permission)
+            } else {
+                println("have $permission  for context")
+            }
+        }
+    }
+}
+
+private fun grantPermission(context: Context, permission: String) {
```

Comment:
@eugene-krivobokov this function works properly and does not show any errors, but permissions are not set -.-